2 small fixes to alo recipes

aphecetche opened the pull request.

dberzano commented on defaults-alo.sh:
```diff
@@ -28,6 +28,8 @@ overrides:
     prefer_system_check: |
       printf "#include \"gsl/gsl_version.h\"\n#define GSL_V GSL_MAJOR_VERSION * 100 + GSL_MINOR_VERSION\n# if (GSL_V < 116)\n#error \"Cannot use system's gsl. Notice we only support versions from 1.16 (included)\"\n#endif\nint main(){}" | gcc  -I$(brew --prefix gsl)/include -xc++ - -o /dev/null
+  AliRoot:
+    tag: "v5-09-02a"
```
We have just released v5-09-03 (validated), maybe you want to adjust that?

aphecetche commented:
Hi Dario,

Changed the AliRoot version. Thanks for the tip.

Concerning the other changes I believe I've fixed them too. Please note however that I started alo.sh from o2.sh which has the same "issues" then.

Regards,

dberzano commented:
Hi Laurent, yes indeed, we'll have to "fix" them. I still do not see the other defaults made according to `defaults-o2.sh`, in particular:

* autotools override
* C++ standard set to C++14

O2 software **must** be compiled with C++14 now. Can you please test & fix? Thanks!

aphecetche commented:
hum, now that's a lot of commits for simple change. Should I just close this PR and open a new one maybe ?

dberzano commented:
Just `git pull --rebase origin master` and you'll be fine :-)

aphecetche commented:
Hum, not quite... Must have messed up sth...

~/alice/o2-ref/alidist   alo-module
❯ git pull --rebase origin master
From https://github.com/alisw/alidist
 * branch            master     -> FETCH_HEAD
First, rewinding head to replay your work on top of it...
Applying: Fix the alo module file
Applying: Fix the aliroot version used to avoid recompiling the master everytime...
Applying: Bump AliRoot version to v5-09-03
Applying: sh to bash, no backticks, builtin brackets
Applying: Use correct casing for pythia and pythia6 modulefiles (#689)
Using index info to reconstruct a base tree...
M       fairroot.sh
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.
Applying: Fix typo resulting in wrong PROTOBUF_ROOT when nanomsg installed (with brew) (#680)
Using index info to reconstruct a base tree...
M       fairroot.sh
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.
Applying: Initial attempt at supporting shuttle (#688)
Using index info to reconstruct a base tree...
M       aliroot.sh
Falling back to patching base and 3-way merge...
Auto-merging aliroot.sh
CONFLICT (content): Merge conflict in aliroot.sh
error: Failed to merge in the changes.
Patch failed at 0007 Initial attempt at supporting shuttle (#688)
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
